src/createApp.ts

```diff
@@ -837,6 +852,48 @@ const createApp = compose({
 			return instanceRegistries.get(this).identifyStore(store);
 		},
 
+		createWidget<U extends Child, O>(this: App, factory: ComposeFactory<U, O>, options: any = {}): Promise<[ string, U ]> {
+			const app = this;
+			const { defaultWidgetStore, registryProvider } = this;
+			const { id = generateWidgetId() } = options;
+			// Like for custom elements, don't add the generated ID to the options.
+
+			function configureWidget(): Promise<any> | void {
+				// Ensure no other widget with this ID exists.
+				if (publicRegistries.get(app).hasWidget(id)) {
+					return Promise.reject(new Error(`A widget with ID '${id}' already exists`));
+				}
+
+				if (!options.registryProvider) {
+					options.registryProvider = registryProvider;
+				}
+
+				if (options.id && !options.stateFrom && defaultWidgetStore) {
```

Comment:
No, my feeling was that if the creator is asserting a `stateFrom` then they know what they are doing and therefore are responsible for the subsequent "magic".  Otherwise we have to find a way to flag the "do not add" when someone expressly doesn't want that behaviour.